# **Predicting Student Performance**
* https://archive.ics.uci.edu/ml/machine-learning-databases/00320/
* Regression model is used to fit data to predict student performance in secondary school
* Explanation of each feature of the datasets is given on the following link: *https*://archive.ics.uci.edu/ml/datasets/Student+Performance


In [ ]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

In [ ]:
data = pd.read_csv('student-mat.csv', sep=';')       # extracted student-mat.csv file from the zip folder. This contains the grades data and the performance prediction can be done now.

In [ ]:
X = data.drop('G3', axis=1)     # G3 = target variable to predict       
y = data['G3']

In [ ]:
X = pd.get_dummies(X, drop_first = True)

In [ ]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size =0.2, random_state = 42)

In [ ]:
model = LinearRegression()
model.fit(X_train, y_train)

LinearRegression()

In [ ]:
y_pred = model.predict (X_test)         # predicting grades for the data

In [ ]:
mse = mean_squared_error (y_test, y_pred)
print ('Mean squared error: ', mse)

Mean squared error:  5.6566428332312215


Per our output, it can be understood that on average, the squared difference between the predicted grades and the actual grades of the test set is 5.6566. 

Well, it suggests that the model is performing reasonably well at predicting the grades of high school students based on the input features, but there may be room for improvement with further experimentation and optimization.